Submitted By: Matt Burgess (matthew at linuxfromscratch dot org)
Date: 2006-10-28
Initial Package Version: 3.2
Upstream Status: From Upstream
Origin: http://ftp.gnu.org/gnu/bash/bash-3.2-patches/bash32-001
Description: When using historical ``-style command substitution, bash
             incorrectly attempts to interpret shell comments while scanning for
             the closing backquote.

diff -Naur bash-3.2.orig/parse.y bash-3.2/parse.y
--- bash-3.2.orig/parse.y	2006-09-19 20:37:21.000000000 +0000
+++ bash-3.2/parse.y	2006-10-27 20:08:59.000000000 +0000
@@ -2735,7 +2735,7 @@
 /* itrace("parse_matched_pair: open = %c close = %c", open, close); */
   count = 1;
   pass_next_character = backq_backslash = was_dollar = in_comment = 0;
-  check_comment = (flags & P_COMMAND) && qc != '\'' && qc != '"' && (flags & P_DQUOTE) == 0;
+  check_comment = (flags & P_COMMAND) && qc != '`' && qc != '\'' && qc != '"' && (flags & P_DQUOTE) == 0;
 
   /* RFLAGS is the set of flags we want to pass to recursive calls. */
   rflags = (qc == '"') ? P_DQUOTE : (flags & P_DQUOTE);
diff -Naur bash-3.2.orig/patchlevel.h bash-3.2/patchlevel.h
--- bash-3.2.orig/patchlevel.h	2006-04-13 12:31:04.000000000 +0000
+++ bash-3.2/patchlevel.h	2006-10-27 20:08:59.000000000 +0000
@@ -25,6 +25,6 @@
    regexp `^#define[ 	]*PATCHLEVEL', since that's what support/mkversion.sh
    looks for to find the patch level (for the sccs version string). */
 
-#define PATCHLEVEL 0
+#define PATCHLEVEL 1
 
 #endif /* _PATCHLEVEL_H_ */
